rir/src/compiler/native/lower_llvm.cpp

```diff
@@ -1497,19 +1501,29 @@ void LowerFunctionLLVM::incrementNamed(llvm::Value* v, int max) {
     builder.SetInsertPoint(done);
 };
 
-void LowerFunctionLLVM::nacheck(llvm::Value* v, BasicBlock* isNa,
+void LowerFunctionLLVM::nacheck(llvm::Value* v, PirType type, BasicBlock* isNa,
                                 BasicBlock* notNa) {
+    assert(type.isA(PirType::num().scalar()));
     if (!notNa)
         notNa = BasicBlock::Create(C, "", fun);
-    if (v->getType() == t::Double) {
-        auto isNotNa = builder.CreateFCmpUEQ(v, v);
-        builder.CreateCondBr(isNotNa, notNa, isNa, branchMostlyTrue);
+    if (!type.maybeNAOrNaN()) {
+        // Don't actually check na because we statically know it's not
+        // Was having trouble using just CreateBr. The conditional jump will get
+        // optimized away
```

Comment:
ok, fine, but maybe at least unify all the 3 `builder.CreateCondBr(isNotNa, notNa, isNa, branchMostlyTrue);` into one and then just set the `isNotNa` to true in this case. Btw the correct constant to pass is `builder.getTrue()`